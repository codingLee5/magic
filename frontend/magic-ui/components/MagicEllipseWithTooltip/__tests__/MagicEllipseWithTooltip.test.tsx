import { render, screen } from "@testing-library/react"
import MagicThemeProvider from "../../ThemeProvider"
import { describe, it, expect } from "vitest"
import MagicEllipseWithTooltip from "../index"

const renderWithTheme = (component: React.ReactElement) =>
	render(<MagicThemeProvider theme="light">{component}</MagicThemeProvider>)

describe("MagicEllipseWithTooltip", () => {
	it("应该正常渲染", () => {
		renderWithTheme(<MagicEllipseWithTooltip text="测试内容" maxWidth="100px" />)
		expect(screen.getByText("测试内容")).toBeInTheDocument()
	})

	// 快照测试
	describe("快照测试", () => {
		it("基础省略文本快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicEllipseWithTooltip text="测试内容" maxWidth="100px" />,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("长文本省略快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicEllipseWithTooltip
					text="这是一个很长的文本内容，应该会被省略显示"
					maxWidth="100px"
				/>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("自定义最大宽度快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicEllipseWithTooltip text="测试内容" maxWidth="200px" />,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("带自定义样式快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicEllipseWithTooltip
					text="测试内容"
					maxWidth="100px"
					style={{ color: "red" }}
				/>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("带类名快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicEllipseWithTooltip
					text="测试内容"
					maxWidth="100px"
					className="custom-ellipse"
				/>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("空文本快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicEllipseWithTooltip text="" maxWidth="100px" />,
			)
			expect(asFragment()).toMatchSnapshot()
		})
	})
})
