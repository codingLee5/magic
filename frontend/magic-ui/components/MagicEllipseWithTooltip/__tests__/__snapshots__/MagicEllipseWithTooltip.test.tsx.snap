// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MagicEllipseWithTooltip > 快照测试 > 基础省略文本快照 1`] = `
<DocumentFragment>
  <div
    aria-describedby="test-id"
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis; max-width: 100px;"
  >
    测试内容
  </div>
</DocumentFragment>
`;

exports[`MagicEllipseWithTooltip > 快照测试 > 带类名快照 1`] = `
<DocumentFragment>
  <div
    aria-describedby="test-id"
    class="custom-ellipse"
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis; max-width: 100px;"
  >
    测试内容
  </div>
</DocumentFragment>
`;

exports[`MagicEllipseWithTooltip > 快照测试 > 带自定义样式快照 1`] = `
<DocumentFragment>
  <div
    aria-describedby="test-id"
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis; max-width: 100px;"
  >
    测试内容
  </div>
</DocumentFragment>
`;

exports[`MagicEllipseWithTooltip > 快照测试 > 空文本快照 1`] = `
<DocumentFragment>
  <div
    aria-describedby="test-id"
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis; max-width: 100px;"
  />
</DocumentFragment>
`;

exports[`MagicEllipseWithTooltip > 快照测试 > 自定义最大宽度快照 1`] = `
<DocumentFragment>
  <div
    aria-describedby="test-id"
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis; max-width: 200px;"
  >
    测试内容
  </div>
</DocumentFragment>
`;

exports[`MagicEllipseWithTooltip > 快照测试 > 长文本省略快照 1`] = `
<DocumentFragment>
  <div
    aria-describedby="test-id"
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis; max-width: 100px;"
  >
    这是一个很长的文本内容，应该会被省略显示
  </div>
</DocumentFragment>
`;
